Style/Hierarchy.md

```diff
@@ -28,15 +31,16 @@ NJOY21 has a fine-grained distribution of source code between source files.
 - For each **name** within an *effective* namespace, a corresponding implementation file (i.e., `.cpp`) is provided with unit tests for each function of that name. For naming conventions for these files, see the [Directory Structure] section below.
```

Comment:
[These lines](https://github.com/njoy/DevelopersGuide/pull/1/files?diff=unified&short_path=04c6e90#diff-84a131c1c1d6b66a5ebf0992e62e2d92R29) should add an exclusion for header only projects